probe_scraper/runner.py

```diff
@@ -220,11 +225,18 @@ def load_glean_metrics(cache_dir, out_dir, repositories_file, dry_run):
                         "subject": "Probe Scraper: Error on Metric Parsing",
                         "message": msg
                     })
+            dependencies[repo_name][commit_hash] = dependency_scraper.parse_dependencies(
```

Comment:
I _think_ the correct thing to do here would be to only check commit hashes on commits where the dependencies file changed. In this case since the dependency file isn't actually in git, we would need the original dependencies file (`build.gradle`?).

Since at this point we're not expecting: many libraries, or many additions/removals of libraries, this may be an optional change, but it's worth noting that we could theoretically miss dependencies with this approach.

Reply:
Ah, that's a good point.  Changes can actually occur across multiple files -- there are multiple `build.gradle` files, and technically, everything in `buildSrc` could affect the dependencies.

I guess we should add a `dependencies_files` key to `repositories.yaml` and treat it similarly to `metrics_files` with respect to how the git commit hashes are determined?

Reply:
I think that's the way. Things do seem to be getting a little bloated, but I'm not sure of a good alternative.